### PR TITLE
[Test] add a test for empty conic problem

### LINKED ISSUE #1741: Add test for conic problem with constraints but no variables
See this issue in ECOS: https://github.com/jump-dev/ECOS.jl/issues/132#issuecomment-1039861056

We should see what happens for models of the form:
```Julia
model = MOI.Utilities.Model{Float64}()
f = MOI.VectorAffineFunction(MOI.VectorAffineTerm{Float64}[], [1.0, 0.5, 0.5])
MOI.add_constraint(model, f, MOI.SecondOrderCone(3))
```

<s>They might solve, or they might error, but they shouldn't segfault.</s>

Test for `MOI.OPTIMAL` or `MOI.INVALID_MODEL` as the termination status.

### PR DESCRIPTION
**Pull request by odow**
Closes #1741

### CODE REVIEW

File: src/Test/test_conic.jl [hunk 1]
```diff
@@ -6791,3 +6791,37 @@ function setup_test(
...
 end
+
+function test_conic_empty_matrix(
+    model::MOI.ModelLike,
+    config::Config{T},
+) where {T}
+    @requires _supports(config, MOI.optimize!)
+    F = MOI.VectorAffineFunction{T}
+    @requires MOI.supports_constraint(model, F, MOI.SecondOrderCone)
+    terms = MOI.VectorAffineTerm{Float64}[]
```
Comment (odow) [inline src/Test/test_conic.jl:6802]:
```suggestion
    terms = MOI.VectorAffineTerm{T}[]
```
